src/components/pages/text-curation/edit/table/form/map-locations.tsx

```diff
@@ -0,0 +1,108 @@
+import "leaflet/dist/leaflet.css";
+
+import { Box, Collapse } from "@chakra-ui/react";
+import { LEAFLET_MARKER_ICON } from "@static/constants";
+import L from "leaflet";
+import { LatLngBoundsExpression } from "leaflet";
+import React from "react";
+import { Marker, Tooltip } from "react-leaflet";
+import { MapContainer } from "react-leaflet";
+import { TileLayer } from "react-leaflet";
+
+const onMarkerClickHandler = (
+  row,
+  suggestion,
+  setLatitude,
+  setLongitude,
+  setLocationAccuracy,
+  hForm,
+  locationRef
+) => {
+  setLatitude(suggestion.coordinates[1]);
+  setLongitude(suggestion.coordinates[0]);
+  setLocationAccuracy(suggestion.locationAccuracy);
+  hForm.setValue("longitude", suggestion.coordinates ? suggestion.coordinates[0] : row.longitude);
+  hForm.setValue("latitude", suggestion.coordinates ? suggestion.coordinates[1] : row.latitude);
+  hForm.setValue(
+    "locationAccuracy",
+    suggestion.locationAccuracy ? suggestion.locationAccuracy : row.locationAccuracy
+  );
+  locationRef.current.onChange(
+    {
+      value: suggestion.label,
+      label: suggestion.label,
+      coordinates: suggestion.coordinates,
+      locationAccuracy: suggestion.locationAccuracy
+    },
+    { name: locationRef.current.props.inputId }
+  );
+};
+
+const MapSuggedtedLocations = ({
+  row,
+  setLatitude,
+  setLongitude,
+  setLocationAccuracy,
+  hForm,
+  locationRef,
+  isOpen
+}) => {
+  const allLatitudes = row.peliasLocations
+    ? row.peliasLocations.map((suggestion) => suggestion.coordinates[1])
+    : [];
+  const allLongitudes = row.peliasLocations
+    ? row.peliasLocations.map((suggestion) => suggestion.coordinates[0])
+    : [];
+  const mapBounds: LatLngBoundsExpression = [
+    [Math.min(...allLatitudes) - 1, Math.min(...allLongitudes) - 1],
+    [Math.max(...allLatitudes) + 1, Math.max(...allLongitudes) + 1]
+  ];
+  return (
+    <Box>
+      <Collapse in={isOpen}>
+        {row.peliasLocations && (
```

Comment:
this will probably be always set to `true` since empty array is also positive?

`row.peliasLocations.length > 0`??